timely_beliefs/tests/test_df_resampling.py

```diff
@@ -241,32 +243,54 @@ def test_percentages_and_accuracy_of_probabilistic_model(df_4323: BeliefsDataFra
 def test_downsample_once_upsample_once_around_dst(
     df_wxyz: Callable[[int, int, int, int, Optional[datetime]], BeliefsDataFrame]
 ):
+    """Fast track resampling is enabled because the data contains 1 deterministic belief per event and a unique belief time and source."""
+    downsampled_event_resolution = timedelta(hours=24)
+    upsampled_event_resolution = timedelta(minutes=10)
     start = pytz.timezone("Europe/Amsterdam").localize(datetime(2020, 3, 29, 0))
     df = df_wxyz(25, 1, 1, 1, start)  # 1 deterministic belief per event
+    df.iloc[0] = np.NaN  # introduce 1 NaN value
     print(df)
-    event_resolution_1 = timedelta(hours=24)
-    df_resampled_1 = df.resample_events(
-        event_resolution_1, keep_only_most_recent_belief=True
-    )  # enables fast track resampling for 1 deterministic belief per event
+
+    # Downsample the original frame
+    df_resampled_1 = df.resample_events(downsampled_event_resolution)
     print(df_resampled_1)
     # todo: uncomment if this is ever fixed: https://github.com/pandas-dev/pandas/issues/35248
-    # assert df_resampled_1.index.get_level_values("event_start")[1] == pd.Timestamp(start) + event_resolution
-    assert len(df_resampled_1) == 2 * event_resolution_1 / event_resolution_1  # 2
+    # assert df_resampled_1.index.get_level_values("event_start")[1] == pd.Timestamp(start) + event_resolution_1
+    assert len(df_resampled_1) == 2  # the data falls in 2 days
     assert df_resampled_1.sensor == df.sensor
-    assert df_resampled_1.event_resolution == event_resolution_1
-    event_resolution_2 = timedelta(minutes=5)
-    df_resampled_2 = df_resampled_1.resample_events(
-        event_resolution_2, keep_only_most_recent_belief=True
+    assert df_resampled_1.event_resolution == downsampled_event_resolution
+    assert df_resampled_1["event_value"].isnull().sum() == 0
+
+    # Upsample the original frame
+    df_resampled_2 = df.resample_events(
+        upsampled_event_resolution, keep_nan_values=True
```

Comment:
Is there a check if the `NaN` value is still in the result?

Reply:
Actually, upon resampling there should be 2 NaN values. The relevant check is in line 273-275.

Reply:
I'll add an inline comment.

Reply:
Still confuse - Are you taking the sum of NaN values here?